docs/shared/workspace/creating-libraries.md

```diff
@@ -0,0 +1,33 @@
+# Creating Libraries
+
+Like a lot of decisions in programming, deciding to make a new Nx library or not is all about trade-offs. Each organization will decide on their own conventions, but here are some trade-offs to bear in mind as you have the conversation.
+
+## Should I Make a New Library?
+
+There are three main benefits to breaking your code up into more libraries.
+
+### 1. Faster Commands
+
+The more granular your libraries are, the more effective `nx affected` and Nx's computation cache will be. For example, if `libraryA` contains 10 tests, but only 5 of them were affected by a particular code change, all 10 tests will be run by `nx affected --target=test`. If you can predict which 5 tests are usually run together, you can split all the related code into a separate library to allow the two groups of 5 tests to be executed independently.
+
+### 2. Visualizing Architecture
+
+The `nx dep-graph` command generates a graph of how apps and libraries depend on each other. If most of your code lives in a few giant libraries, this visualization doesn't provide much value.
+
+### 3. Enforcing Constraints
+
+You can enforce constraints on how different types of libraries depend on each other [using tags](/{{framework}}/workspace/structure/monorepo-tags). Following pre-determined conventions on what kind of code can go in different types of libraries allows your tagging system to enforce good architectural patterns.
+
+Also, each library defines its own API, which allows for encapsulating logic that other parts of codebase can not access.
```

Comment:
Possibly related: ability to assign 'code ownership' of specific libs to a given team/developer